src/hotspot/share/ci/ciEnv.cpp

```diff
@@ -1773,3 +1774,11 @@ void ciEnv::dump_inline_data(int compile_id) {
     }
   }
 }
+
+bool ciEnv::is_multifield_scalarized(BasicType bt, int vec_length) {
+#if COMPILER2
+  return InlineTypeNode::is_multifield_scalarized(bt, vec_length);
+#else
+  return false;
```

Comment:
1. I think it's better to implement `is_multifield_scalarized()` here, and reference it in C2 compiler. It's opposite here.
2. By default, it should return `true` instead of `false` if `COMPILER2` is not supported.

Reply:
Agree with 2, thanks, will correct it! Idea here is that ci should leverage compiled exposed interfaces to determine scalarizability.

Reply:
What is the expected return value of `is_multifield_scalarized()` for C1 compiler?

Reply:
> What is the expected return value of `is_multifield_scalarized()` for C1 compiler?

C2 is the only runtime component which vectorizes multifield bundle and lower tier compiler should return a true value.

Reply:
Yes, correct. Per my understanding `#if COMPILER2` is a compile flag which is true if C2 is not disabled. Hence this code is just like directly `return InlineTypeNode::is_multifield_scalarized(bt, vec_length);`, right? And then it returns the same value for interpreter/c1/c2.

Reply:
I think what we need is adding an interface here, and implement it in different compilers, or a runtime check like https://github.com/openjdk/valhalla/blob/lworld%2Bvector/src/hotspot/share/compiler/compilerDefinitions.inline.hpp#L49. WDYT?

Reply:
> Yes, correct. Per my understanding `#if COMPILER2` is a compile flag which is true if C2 is not disabled. Hence this code is just like directly `return InlineTypeNode::is_multifield_scalarized(bt, vec_length);`, right? And then it returns the same value for interpreter/c1/c2.

My bad, I completely overlooked it, you are correct.